p4-16/spec/P4-16-draft-spec.mdk

```diff
@@ -777,78 +756,75 @@ The VSS architecture provides an incremental checksum extern block,
+features provided by the architecture---e.g., recirculation---but it does use
+preprocessor ```#include``` directives (see Section [#sec-preprocessor]).
 
 ~ Figure { #fig-vssmau; caption: "Diagram of the match-action pipeline expressed by the VSS P4 program." }
 ![vssmau]
 ~
 [vssmau]: figs/vssmau.png { width: 100%; page-align: here }
 
 []{tex-cmd: "\indent"}
-Parsing attempts to recognize an Ethernet header and an IPv4 header;
-if these headers are missing parsing terminates with an
+The parser attempts to recognize an Ethernet header followed by an IPv4 header.
+If either of these headers are missing, parsing terminates with an
 error. Otherwise it extracts the information from these headers into a
-structure with type ```Parsed_packet```. The match-action pipeline is
-shown in Figure [#fig-vssmau]; it comprises 4 match-action units
+```Parsed_packet``` structure. The match-action pipeline is
+shown in Figure [#fig-vssmau]; it comprises four match-action units
 (represented by the P4 ```table``` keyword):
 
-- If any parser error has occurred, the packet is dropped (sending it
+- If any parser error has occurred, the packet is dropped (i.e., by assigning ```outputPort```
   to ```DROP_PORT```)
-- The first table uses the IPv4 destination address to discover the ```outputPort```
-  and the IPv4 address of the next hop. If this
-  look-up fails, the packet is dropped. The table also decrements the
-  IPv4 ttl value.
-- The second table checks the ttl value: if the ttl becomes 0, the
-  packet is sent through the CPU port towards the control plane.
-- The third table uses the IPv4 address of the next hop (computed by
-  the first table) to figure out the Ethernet address of the next hop.
+- The first table uses the IPv4 destination address to determine the ```outputPort```
+  and the IPv4 address of the next hop. If this lookup fails, the packet is dropped.
+  The table also decrements the IPv4 ```ttl``` value.
+- The second table checks the ```ttl``` value: if the ```ttl``` becomes 0, the
+  packet is sent to the control plane through the CPU port.
+- The third table uses the IPv4 address of the next hop (which was computed by
+  the first table) to determine the Ethernet address of the next hop.
 - Finally, the last table uses the ```outputPort``` to identify the
   source Ethernet address of the current switch, which is set in the
   outgoing packet.
 
-The deparser puts together a packet by reassembling the Ethernet and
-IPv4 headers as computed in the pipeline.  This example uses
-preprocessor ```#include``` directives (see Section
-[#sec-preprocessor]).
+The deparser constructs the outgoing packet by reassembling the Ethernet and
+IPv4 headers as computed by the pipeline.
 
 ~ Begin P4Example
+// Include P4 core library
 # include <core.p4>
 
-// include the very simple switch declaration from the previous section
+// Include very simple switch architecture declarations
```

Comment:
You changed the file name above but not here.
Note that all our P4 architecture description files are named with _model.p4 in the actual implementation, so I prefer this name.